xc/device.ml

```diff
@@ -1444,22 +1461,34 @@ module Vusb = struct
       end
     in
     let is_running = Qemu.is_running ~xs domid in
-    match is_running with
-    | true  ->
-      let cmd = Qmp.(Device_add
-                       ( "usb-host"
-                       , id
-                       , Some (get_bus version, hostbus, hostport)
-                       ))
-      in qmp_send_cmd domid cmd |> ignore
-    | _ -> ()
-
-  let vusb_unplug ~xs ~domid ~id =
+    if is_running then
+      begin
+        (* Need to reset USB device before passthrough to vm according to CP-24616.
+           Also need to do deprivileged work in usb_reset script if QEMU is deprivileged.
+        *)
+        begin match Qemu.pid ~xs domid with
+          | Some pid -> call_usb_reset hostbus hostport domid pid "attach" priv
+          | _ -> failwith (Printf.sprintf "qemu pid does not exist for vm %s" (string_of_int domid))
```

Comment:
If this is raised, the subsequent `qmp` command `Device_add` will not be issued. Is this correct?

Reply:
If the pid does not exists, usb_reset.py shouldn't be called successfully. The pid is used when setting up cgroup, so I think the failure raise is correct and passthrough won't be success.

Reply:
Also, why we prefer `failwith` to something more tractable like `raise (Internal_error ...`